### PR TITLE
[LENS-415] Styled slider input

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

- Styles <Slider /> component to provide a rich branded experience 

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [ ] PR is ideally < 400LOC
- [ ] Link(s) to related Github issues

### :camera: Screenshots

Screen Shot 2019-12-03 at 5 22 54 PM

### CODE REVIEW

**Comment by dbchristopher** [comment]
@mdodgelooker 😓 wow this customized Slider component turned out to be a lot of work. I think it's all good to go now! I'm still planning to update the documentation related to these changes, but I'll do that in a followup PR. If you have any time to look at this tomorrow it would be greatly appreciated 🙏

File: packages/components/src/Button/IconButton.test.tsx [hunk 1]
```diff
@@ -26,14 +26,13 @@
...
 import React from 'react'
+import noop from 'lodash/noop'
 import { theme } from '@looker/design-tokens'
 import { assertSnapshot } from '@looker/components-test-utils'
 import { ThemeProvider } from 'styled-components'
 import { render, fireEvent } from '@testing-library/react'
 import { IconButton } from './IconButton'
 
-const noop = () => {}
-
```
Comment (dbchristopher) [inline packages/components/src/Button/IconButton.test.tsx]: randomly started getting these lint errors, so switched to lodash's `noop` function instead:
```
 /Users/danielchristopher/Looker/components/packages/components/src/List/ListItem.test.tsx
  32:20  error  Unexpected empty arrow function  @typescript-eslint/no-empty-function
```

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 2]
```diff
@@ -24,13 +24,325 @@
...
+    border: 3px solid
+      ${branded ? colors.semanticColors.primary.main : colors.palette.blue500};
+    height: ${knobSize}px;
+    width: ${knobSize}px;
+    background: ${colors.palette.white};
+    ${isFocused &&
+      `box-shadow: 0 0 6px ${
+        branded ? colors.semanticColors.primary.dark : colors.palette.blue700
+      };`}
+  `}
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: renders focus style during keyboard navigation, but clears focus style on click
focus-styles

File: packages/playground/src/Slider/SliderDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+import React, {
+  useState,
+  SyntheticEvent,
+  Dispatch,
+  SetStateAction,
+} from 'react'
+import { Card, CardContent, Slider, Heading } from '@looker/components'
+import styled from 'styled-components'
+
+export const SliderDemo = () => {
```
Comment (dbchristopher) [inline packages/playground/src/Slider/SliderDemo.tsx]: Going to copy most of these examples over to mdx documentation after this PR is reviewed and merged

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 4]
```diff
@@ -24,13 +24,325 @@
...
+    ${sliderThumbCss}
+  }
+
+  &::-ms-track {
+    width: 100%;
+    cursor: pointer;
+    background: transparent;
+    border-color: transparent;
+    color: transparent;
+  }
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: I'd love to double check how IE is handling these styles before this goes out. @mdodgelooker what are you using for cross-browser testing these days?

Comment (dbchristopher) [reply packages/components/src/Form/Inputs/Slider/Slider.tsx]: @mdodgelooker I'm just not getting anywhere with running windows on my machine. would you mind taking a look at this in edge to see if it's rendering correctly?

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 5]
```diff
@@ -24,13 +24,335 @@
...
+export interface SliderProps
+  extends SpaceProps,
+    WidthProps,
+    Omit<InputProps, 'type'>,
+    SliderSizeProps {
+  'aria-labelledby'?: string
+  branded?: boolean
+  max?: number
+  min?: number
+  ref?: Ref<HTMLInputElement>
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: This is duplicated on line 50 – the common pattern (and what we've been using) seems to be _not_ adding ref to the props interface, and instead... (see below)

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 6]
```diff
@@ -24,13 +24,335 @@
...
+  step?: number
+  value?: number
+}
 
-export const Slider = styled.input.attrs({ type: 'range' })<SliderProps>`
+const SliderInternal: FC<SliderProps> = forwardRef(
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: Remove `: FC<SliderProps>` and...

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 7]
```diff
@@ -24,13 +24,335 @@
...
+      value = 0,
+      step,
+      onChange,
+      size = 'medium',
+      name,
+      id,
+      className,
+      disabled,
+      ...restProps
+    },
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx]: Add `: SliderProps` after the `}` here.

**Comment by dbchristopher** [comment]
ok updated with those changes @mdodgelooker!

**Pull request merged**